### DISCUSSION TITLE
RTH Climb alternative

### DISCUSSION THREAD

**Discussion by A2BELE** [iNavFlight/inav#7237] [Ideas]
As many of us have have had the issue of quads shooting to the moon when Failsafe is activated.
( can be due to numerous reasons, but this often ends up with quads hundreds of meters in the air which is not safe. ) 

 I have a program change suggestion of modify RTH climb program so that it can only climb at a fixed rated. Instead of climbing to an altitude climb for X seconds with motors at X speed. Say 10 secs at 2000 or something.

Just an Idea that could end the constant rockets.

**Comment by paul-fornage** [comment]
This seems like a good idea, but I think climbing at a fixed speed, like VNAV would be better, to account for quads with different power to weight ratios. iNav could also theoretically look at a black box and determine what throttle position corresponds to what speed, but that would be a huge undertaking.

Also when I first read that I was confused because 2000 is servo PWM for full throttle Lol.

**Reply by A2BELE** [reply]
Hi Paul,
I was more thinking the X would be a user adjusted value with inav configuration so then we could customize it. 
I guess after a bit of trial and error some defaults might be able to be applied.

I program plcs and this is what I would do if I knew how to program for inav.

**Comment by avsaase** [comment]
The quad shooting to the moon problem is almost always caused by a problem with the altitude estimation. The quad want to climb to the RTH altitude but for some reason it estimates the altitude is decreasing (negative vertical speed) instead if increasing (positive vertical speed). This causes the altitude PID to freak out and the quad becomes a rocket. On my quad vibrations messed with the accelerometer which messed with the altitude estimation. Please record a blackbox log to see this describes your problem. Look at the target and actual altitude and vertical speed.

**Reply by A2BELE** [reply]
Of course, but the idea behind failsafe is to failsafe and not create a hazard which happens when your quad is 300m over your head.
I can setup my quad and the althold , rth is perfect. The quad doesn’t move when I push it with a stick. Inav alt & pos hold works very very well setup.

what I’m talking about is when one little thing is out of whack you end up with a rocket. This is just another way of how this could be addressed.

**Reply by avsaase** [reply]
Sure, we could add some runaway detection.

**Reply by A2BELE** [reply]
Yeah that would be another option, although the issue happens so fast currently that it would be better to maybe tackle it another way.

**Reply by avsaase** [reply]
I disagree with that. We have this system of altitude control for RTH/failsafe specially to allow the pilot to set a procedure that is safe and  for their flying environment. Replacing it with a fixed throttle level on a timer makes it very hard to predict what altitude will be reached as this depends entirely on your quad.

As a side note, if RTH works normally then failsafe should work the same AFAIK. I can't check the code right now to confirm that though.

**Reply by A2BELE** [reply]
Was just an idea, I’m an electrician that specialises on machinery and programming. To me Fail safe is exactly that, from my experience and obviously a lot of others who have had rockets there is something that needs a tweak as it doesn’t fail safe that’s all.
I don’t know how the code works for the alt hold and RTH. 

I actually just went for a fly in 25knot winds, ran through all the tests and everything worked perfect. ( Bobbed up and down a bit more then normal because if the wind )
